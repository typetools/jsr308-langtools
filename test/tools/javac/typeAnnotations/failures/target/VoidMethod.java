/*
 * @test /nodynamiccopyright/
 * @bug 6843077
 * @summary test invalid location of TypeUse
 * @author Mahmood Ali
 * @compile/fail/ref=VoidMethod.out -XDrawDiagnostics -source 1.8 VoidMethod.java
 */

import java.lang.annotation.Target;
import java.lang.annotation.ElementType;

class VoidMethod {
  @A void test() { }
  // The following is legal:
  @B void test2() { }
}

@Target(ElementType.TYPE_USE)
@interface A { }

@Target({ElementType.TYPE_USE, ElementType.METHOD})
@interface B { }
