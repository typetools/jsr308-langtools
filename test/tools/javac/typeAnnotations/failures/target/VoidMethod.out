VoidMethod.java:35:3: compiler.err.annotation.type.not.applicable
1 error
